Description: Use bash explicitly for tests, to work-around FTBFS
Bug-Debian: http://bugs.debian.org/632494
Bug-Ubuntu: https://bugs.launchpad.net/bugs/831112
Author: Matthias Klose <doko@ubuntu.com>

--- darcs-2.5.2.orig/tests/issue458.sh
+++ darcs-2.5.2/tests/issue458.sh
@@ -16,7 +16,7 @@ cd temp

 mkdir repo1
 darcs initialize --repodir repo1
-printf >repo1/x '#!/bin/sh\ntrue'   # make a shebang'd script
+printf >repo1/x '#!/bin/bash\ntrue'   # make a shebang'd script
 darcs record --repodir repo1 -lam x x
 umask 077                       # DENY ALL access to group, all
 darcs get --set-scripts-executable repo1 repo2
--- darcs-2.5.2.orig/tests/issue885_get_to_match.sh
+++ darcs-2.5.2/tests/issue885_get_to_match.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash

 # Issue885: Regression: "darcs get --to-match" does not work anymore under 2.0

--- darcs-2.5.2.orig/tests/issue27.sh
+++ darcs-2.5.2/tests/issue27.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash

 set -ev

--- darcs-2.5.2.orig/tests/issue1269_setpref_predist.sh
+++ darcs-2.5.2/tests/issue1269_setpref_predist.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash

 set -ev

--- darcs-2.5.2.orig/tests/send-external.sh
+++ darcs-2.5.2/tests/send-external.sh
@@ -18,7 +18,7 @@ darcs add foobar
 darcs rec -a -m add-foobar

 cat > saveit.sh <<EOF
-#!/bin/sh
+#!/bin/bash
 # send-mail1.sh: Test sendmail command for darcs send 1
 # 2008-Oct-06 22.25 / TN
 set -ev
--- darcs-2.5.2.orig/tests/hidden_conflict.sh
+++ darcs-2.5.2/tests/hidden_conflict.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash

 set -ev

--- darcs-2.5.2.orig/tests/failing-issue1317_list-options_subdir.sh
+++ darcs-2.5.2/tests/failing-issue1317_list-options_subdir.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash

 # Test for issue1317 - darcs mv --list-options returns results from root, and
 # not from the current directory
--- darcs-2.5.2.orig/tests/nfs-failure.sh
+++ darcs-2.5.2/tests/nfs-failure.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash

 set -ev

--- darcs-2.5.2.orig/tests/issue525_amend_duplicates.sh
+++ darcs-2.5.2/tests/issue525_amend_duplicates.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash

 set -ev

--- darcs-2.5.2.orig/tests/record_editor.sh
+++ darcs-2.5.2/tests/record_editor.sh
@@ -49,7 +49,7 @@ touch file.t
 darcs add file.t
 touch '; test-command'
 echo > test-command << FOO
-#!/bin/sh
+#!/bin/bash
 echo EVIL
 FOO
 chmod u+x test-command
--- darcs-2.5.2.orig/tests/failing-nice-resolutions.sh
+++ darcs-2.5.2/tests/failing-nice-resolutions.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash

 set -ev

--- darcs-2.5.2.orig/tests/issue1162_add_nonexistent_slash.sh
+++ darcs-2.5.2/tests/issue1162_add_nonexistent_slash.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash

 set -ev

--- darcs-2.5.2.orig/tests/failing-issue1406.sh
+++ darcs-2.5.2/tests/failing-issue1406.sh
@@ -36,7 +36,7 @@ darcs record    --repo R -am 'Move d/f t
 darcs push      --repo R S -a   # Try to push patches between repos.
 darcs push      --repo S R
 rm -rf R/ S/                    # Clean up after ourselves.
-#!/bin/sh
+#!/bin/bash
 set -ev

 . ../tests/lib
--- darcs-2.5.2.orig/tests/issue538.sh
+++ darcs-2.5.2/tests/issue538.sh
@@ -11,7 +11,7 @@ fi

 function make_repo_with_test {
     mkdir temp1 ; cd temp1 ; darcs init
-    echo "#!/bin/sh" > test.sh
+    echo "#!/bin/bash" > test.sh
     echo "echo 'hello world'" >> test.sh
     darcs add test.sh
     darcs record --author=test@test -am test
@@ -95,9 +95,9 @@ cd ..
 # check trackdown with files that become scripts during trackdown
 rm -rf temp1
 mkdir temp1 ; cd temp1 ; darcs init
-echo "#!/bin/sh" > test.sh
+echo "#!/bin/bash" > test.sh
 echo "./helper.sh" >> test.sh
-echo "#!/bin/sh" > helper.sh
+echo "#!/bin/bash" > helper.sh
 echo "echo 'helper speaking'" >> helper.sh
 darcs add test.sh
 darcs add helper.sh
--- darcs-2.5.2.orig/tests/issue709_pending_look-for-adds.sh
+++ darcs-2.5.2/tests/issue709_pending_look-for-adds.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 set -ve

 . lib
